infra/modules/azure_monorepo_single_env_starter_pack/github_branch_rules.tf

```diff
@@ -0,0 +1,25 @@
+resource "github_branch_default" "main" {
+  repository = github_repository.this.name
+  branch     = "main"
+}
+
+resource "github_branch_protection" "main" {
+  repository_id = github_repository.this.name
+  pattern       = "main"
+
+  required_status_checks {
+    strict   = false
```

Comment:
why false?

Reply:
I don't see any value in having the branch up to date before merging as CI pipelines run over the merge commit

Reply:
can you elaborate a little bit more?

Reply:
As we're talking about monorepository (aka a single, potentially large repository), there is an high possibility that PR's branch is not up to date with `main` as other people are working on the same code base. I don't see this as an issue because the delta does not affect the validity of CI checks nor "manual" review as changes are on different paths. On the other hand, if changes are on the same files it is likely to have conflicts and PR cannot be merged

Reply:
let's see what @pagopa/engineering-team-devex thinks about this